File: frontend/src/types/life.ts
// ============================================================
// 生活记录类型定义（二期 M9）
// ============================================================

export interface Habit {
  id: string
  name: string
  icon: string | null
  color: string | null
  frequency: 'daily' | 'weekly' | 'custom'
  target_per_week: number
  reminder_time: string | null
  goal_days: number | null
  status: 'active' | 'paused' | 'archived'
  created_at: string
  updated_at: string
}

export interface HabitCheckin {
  id: string
  habit_id: string
  checkin_date: string
  note: string | null
  created_at: string
}

export interface MoodLog {
  id: string
  score: number  // 1-5
  tags: string[]
  content: string | null
  logged_date: string
  created_at: string
}

export interface Diary {
  id: string
  dimension: string | null  // 生活/学习/工作/情感
  title: string | null
  content: string | null
  diary_date: string
  tags: string[]
  created_at: string
  updated_at: string
}

File: backend/app/schemas/life.py
# ============================================================
# 生活记录 schema（二期 M9）
# ============================================================
from __future__ import annotations

from datetime import date, datetime, time
from pydantic import BaseModel, Field


class HabitOut(BaseModel):
    id: str
    name: str
    icon: str | None = None
    color: str | None = None
    frequency: str = "daily"
    target_per_week: int = 7
    reminder_time: time | None = None
    goal_days: int | None = None
    status: str = "active"
    created_at: datetime
    updated_at: datetime


class CreateHabitRequest(BaseModel):
    name: str = Field(min_length=1, max_length=100)
    icon: str | None = None
    color: str | None = None
    frequency: str = "daily"
    target_per_week: int = 7
    reminder_time: time | None = None
    goal_days: int | None = None


class HabitCheckinOut(BaseModel):
    id: str
    habit_id: str
    checkin_date: date
    note: str | None = None
    created_at: datetime


class MoodLogOut(BaseModel):
    id: str
    score: int
    tags: list = []
    content: str | None = None
    logged_date: date
    created_at: datetime


class CreateMoodLogRequest(BaseModel):
    score: int = Field(ge=1, le=5)
    tags: list = []
    content: str | None = None
    logged_date: date | None = None


class DiaryOut(BaseModel):
    id: str
    dimension: str | None = None
    title: str | None = None
    content: str | None = None
    diary_date: date
    tags: list = []
    # 身份轴（三期 B）
    identity_id: str | None = None
    created_at: datetime
    updated_at: datetime


class CreateDiaryRequest(BaseModel):
    dimension: str | None = None
    title: str | None = None
    content: str | None = None
    tags: list = []
    # 身份轴（三期 B）：可空 = 未归类
    identity_id: str | None = None
    diary_date: date | None = None
